src/transformers/trainer.py

```diff
@@ -836,12 +860,13 @@ def _maybe_log_save_evalute(self, tr_loss, model, trial, epoch):
             )
             self._logging_loss_scalar = tr_loss_scalar
 
-            self.log(logs)
+            self.log("train", logs)
```

Comment:
This will break the `NotebookProgressCallback` at least, and perhaps other things. Really not in favor of renaming things inside the main Trainer as I said earlier.

Reply:
The issue there is that there is no way to know if we are training or evaluating in the log function, so I won't be able to do anything for the TensorBoard or Wandb integration.

Reply:
I don't understand that part. Except for the final evaluation, we're always in the middle of training.

Reply:
Not for the majority of metrics, but for example total_flos and epoch are added "manually" in the log() function, and they are not prefixed by "train", "eval" or anything, so there is a name collision there. And if somebody add other stuff in the log function, the collision will affect more data.

Reply:
Fixed in https://github.com/huggingface/transformers/pull/7935/commits/d57d45fe6246e4d6f85a7dc683858546da569e18